test/junit/scala/tools/nsc/backend/jvm/opt/InlinerTest.scala

```diff
@@ -991,4 +991,26 @@ class InlinerTest extends ClearAfterClass {
     assert(2 == t.collect({case Ldc(_, "hai!") => }).size)     // twice the body of f
     assert(1 == t.collect({case Jump(IFNONNULL, _) => }).size) // one single null check
   }
+
+  @Test
+  def inlineIndyLambda(): Unit = {
+    val code =
+      """class C {
+        |  @inline final def m(s: String) = {
+        |    val f = (x: String) => x.trim
+        |    f(s)
+        |  }
+        |  def t = m("foo")
```

Comment:
So, this is the `destinationClass == calleeDeclarationClass` case, right? Could you also add a test for the inter-class case?